site/docs/v1/tech/guides/breached-password-detection.adoc

```diff
@@ -0,0 +1,274 @@
+
+They will not be able to complete a registration without choosing a more secure password.
+
+If you enable breach detection on login, but do not require a password change, the end user will see no difference between signing in with a secure or compromised password. If you enable sending an email, the user will receive that notice.
+
+If you, on the other hand, require a password change, a user signs in with a password that is found to have been compromised, they'll see:
+
+image::guides/breached-password-detection/oauth-login-breached-message.png[What a user who is signing in with a breached password will see.,width=700,role=box-shadow margin-right]
+
+These pages can be customized, both in the messages displayed to the end user and in the look and feel. Please visit the link:../themes[themes documentation] for more information.
+
+=== API Responses
+
+When you create a user with the link:../apis/users[User APIs] and breached password detection is enabled, you'll receive an error if the password is compromised. The HTTP status code will be `400` and you'll receive a response containing an error object.
+
+.JSON response when attempted user created with a breached password
+[source,json]
+----
+{
+  "fieldErrors": {
+    "user.password": [
+      {
+        "code": "[breachedCommonPassword]user.password",
+        "message": "The [user.password] property value has been breached and may not be used, please select a different password."
+      }
+    ]
+  }
+}
+----
+
+If you enable breach detection on login and don't require a password change, you'll receive a `200` HTTP status code on successful login and the normal JSON response. Please see the link:../apis/login[Login API] documentation for more.
+
+If you require a password change, you'll receive a `203` HTTP status code should the user's password have been compromised. You'll also receive JSON explaining why the password change is required:
+
+.JSON response when attempted user created with a breached password
+[source,json]
+----
+{
+  "changePasswordId": "yo0FiR_y7Rrtrk2vKe_F93PQ-nWljfGGWexgNSbHfXQ",
+  "changePasswordReason": "Breached"
+}
+----
+
+You can then use the link:../apis/users#change-a-users-password[Change a User's Password API] to, well, change the user's password.
+
+== How It Works
+
+Here's a high level diagram of the FusionAuth breached password detection service:
+
+image::guides/breached-password-detection/breached-password-detection-diagram.png[Diagram of the breached password detection system.,width=700,role=box-shadow margin-right]
+
+New breached passwords are ingested and processed by FusionAuth on an ongoing basis. An additional encryption key is used to securely communicate with the FusionAuth Reactor service using a strong AES cipher. 
+
+You can rotate this key by navigating to the [breadcrumb]#Reactor# tab and clicking the *Regenerate Key* button. When you do so, you'll be prompted to confirm this decision:
```

Comment:
While this could be used to rotate the key, the intent was to allow them to modify if it they think it could be compromised. 

Perhaps if we think rotation is important we just need to build it into our end. 

In that context, perhaps this should be discussed as a troubleshooting technique.

Reply:
I changed it to "You can modify this key by navigating to"

I added this sentence as well:

You'll want to do this if you there was any chance the key was compromised, if directed to do so by FusionAuth support, or if you are troubleshooting your Reactor service connection.